pkg/lib/usage/limit.go

```diff
@@ -38,35 +78,31 @@ func (l *Limiter) getResetTime(c *config.UsageLimitConfig) time.Time {
 }
 
 func (l *Limiter) Reserve(name LimitName, config *config.UsageLimitConfig) (*Reservation, error) {
+	return l.ReserveN(name, 1, config)
+}
+
+func (l *Limiter) ReserveN(name LimitName, n int, config *config.UsageLimitConfig) (*Reservation, error) {
 	enabled := config.Enabled != nil && *config.Enabled
 	if !enabled {
-		return &Reservation{taken: false, name: name, config: config}, nil
+		return &Reservation{taken: 0, name: name, config: config}, nil
 	}
 
 	quota := config.Quota
 	key := redisLimitKey(l.AppID, name)
 
+	pass := false
 	tokens := int64(0)
 	err := l.Redis.WithConn(func(conn redis.Redis_6_0_Cmdable) error {
 		ctx := context.Background()
-		usage, err := conn.IncrBy(ctx, key, 1).Result()
-		if err != nil {
-			return err
-		}
-
-		tokens = int64(quota) - usage
-
-		resetTime := l.getResetTime(config)
-		// Ignore error
-		_, _ = conn.PExpireAt(ctx, key, resetTime).Result()
```

Comment:
We used to use PEXPIREAT instead of EXPIREAT. Did we make a mistake here?

Reply:
PEXPIREAT uses millisecond precision and EXPIREAT uses second precision. Since the reset time is always truncated to days, the behavior is same.

Reply:
I get it. You call `t.Unix()` before passing the timestamp to the Lua script.